src/WebUI/Services/TranslationManager.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Ninject;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace WebUI.Services
+{
+    class TranslationManager
+    {
+        private static volatile TranslationManager manager;
```

Comment:
`volatile` is unnecessary in presense of `lock`'s. It only removes single-thread oriented compiler optimizations.
In general, I would omit locks using Nested-Class Singleton or Half-Lazy Singleton (as we have no other static methods). Details [here](http://csharpindepth.com/Articles/General/Singleton.aspx).